core/src/main/java/org/renjin/primitives/S3.java

```diff
@@ -550,15 +550,15 @@ private GenericMethod findNext(Environment methodTable, String name, String clas
         return new GenericMethod(this, method, className, (Function) function);
         
       } else if(methodTable != null && methodTable.hasVariable(method)) {
-        return new GenericMethod(this, method, className, (Function) methodTable.getVariable(method).force(context));
+        return new GenericMethod(this, method, className, (Function) methodTable.getVariable(context, method).force(context));
       
       } else {
         return null;
       }
     }
 
     private Environment getMethodTable() {
-      SEXP table = definitionEnvironment.getVariable(METHODS_TABLE).force(context);
+      SEXP table = definitionEnvironment.getVariable(context, METHODS_TABLE).force(context);
```

Comment:
The idea that the S3 methods table would have active bindings is terrifying!! I almost wonder if it is worth adding a special case flag to prevent the method table from being modified by users.